apps/webapp/app/routes/_app.orgs.$organizationSlug.projects.$projectParam.triggers_.webhooks.$triggerParam._index/route.tsx

```diff
@@ -0,0 +1,180 @@
+
+  if (!trigger) {
+    throw new Response("Trigger not found", {
+      status: 404,
+      statusText: "Not Found",
+    });
+  }
+
+  return typedjson({ trigger });
+};
+
+const schema = z.object({
+  jobId: z.string(),
+});
+
+export const action: ActionFunction = async ({ request, params }) => {
+  const userId = await requireUserId(request);
+  const { organizationSlug, projectParam, triggerParam } = TriggerSourceParamSchema.parse(params);
+
+  const formData = await request.formData();
+  const submission = parse(formData, { schema });
+
+  if (!submission.value) {
+    return json(submission);
+  }
+
+  try {
+    const service = new ActivateSourceService();
+
+    const result = await service.call(triggerParam);
+
+    return redirectWithSuccessMessage(
+      externalTriggerPath({ slug: organizationSlug }, { slug: projectParam }, { id: triggerParam }),
+      request,
+      `Retrying registration now`
+    );
+  } catch (error: any) {
+    return json({ errors: { body: error.message } }, { status: 400 });
+  }
+};
+export const handle: Handle = {
+  //this one is complicated because we render outside the parent route (using triggers_ in the path)
+  breadcrumb: (match, matches) => {
+    const data = useTypedMatchData<typeof loader>(match);
+    if (!data) return null;
+
+    const org = useOrganization(matches);
+    const project = useProject(matches);
+
+    return (
+      <Fragment>
+        <BreadcrumbLink to={projectTriggersPath(org, project)} title="Triggers" />
+        <BreadcrumbIcon />
+        <BreadcrumbLink to={projectWebhookTriggersPath(org, project)} title="Webhook Triggers" />
+        <BreadcrumbIcon />
+        <BreadcrumbLink
+          to={trimTrailingSlash(match.pathname)}
+          title={data.trigger.key}
+        />
+      </Fragment>
+    );
+  },
+};
+
+export default function Page() {
+  const { trigger } = useTypedLoaderData<typeof loader>();
+  const organization = useOrganization();
+  const project = useProject();
+  const navigation = useNavigation();
+  const lastSubmission = useActionData();
+
+  const [form, { jobId }] = useForm({
+    id: "trigger-registration-retry",
+    // TODO: type this
+    lastSubmission: lastSubmission as any,
+    onValidate({ formData }) {
+      return parse(formData, { schema });
+    },
+  });
+
+  const isLoading = navigation.state === "submitting" && navigation.formData !== undefined;
+
+  return (
+    <>
+      <Paragraph variant="small" spacing>
+        Webhook Triggers need to be registered with the external service. You can see the list
+        of attempted registrations below.
+      </Paragraph>
+
+      {!trigger.active &&
+        <Form method="post" {...form.props}>
+        <Callout variant="error" className="justiy-between mb-4 items-center">
+          <Paragraph variant="small" className={cn(variantClasses.error.textColor, "grow")}>
+            Registration hasn't succeeded yet, check the runs below.
+          </Paragraph>
+          {/* <input
```

Comment:
Removed the Registration Retry button as this page shows Registration Runs across all Environments. Might need a UI tweak to display the Retry button on failed Run rows instead. (Will have to make minor changes to the backend too.)